packages/docusaurus-types/package.json

```diff
@@ -13,11 +13,17 @@
     "directory": "packages/docusaurus-types"
   },
   "license": "MIT",
+  "scripts": {
+    "test": "tsc -p ."
```

Comment:
Is this supposed to catch the error? Because I tried locally and it doesn't.

Actually trying to find a solution to catch such mistakes 🤪 let me know if you find one

Reply:
Oh I see what happened. I tested a custom `tsconfig.json` file, then I realized there was a shared config and used `extends`. But the TS server didn't reboot and kept the former config, thus I thought the new config would still catch those errors.

And the new config brought `"skipLibCheck": true` which caused declaration files to be ignored. But if I revert that, al those `@types/webpack` error spawn.

Reply:
Those webpack errors come from conflicting `tapable` definitions. `@types/webpack@4.41.29` depends on `@types/tapable@^1" while `webpack@5.40.0` depends on `tapable@^2` which provides its own definitions. 

Not sure what is meant by "@types/webpack and webpack/types.d.ts are not the same thing" :sweat:  (in root tsconfig); but either upgrading @types/webpack to ^5 or sticking with TS types provided by webpack might fix the issue...

Those TS errors in node_modules are maddening!

Reply:
Previously Webpack used definitively typed (4.x); but it started to ship official typedefs in 5.x. The initial 5.x typedefs were missing a lot of things, I think they completed support more recently so we can probably try again to remove `@types/webpack` once for all